fails:The throw keyword clears the current exception
